Tensile/Common.py

```diff
@@ -2232,36 +2246,47 @@ def gfxName(arch):
     name = str(arch[0]) + str(arch[1]) + ('%x' % arch[2])
     return 'gfx' + ''.join(map(str,name))
 
+
+def detectIsaWindows(output):
+    i = 0
+    for line in output:
+      if 'gcnArchName' in line:
+        arch = gfxArch(line.split()[1].strip())
+        if arch and arch in globalParameters["SupportedISA"]:
+          tPrint(1, f"# Detected GPU {i}:    {gfxName(arch)}")
+          globalParameters["CurrentISA"] = arch
+          i += 1
+
+
+def detectIsaLinux(output):
+    for i, line in enumerate(output):
+      arch = gfxArch(line.strip())
+      if arch and arch in globalParameters["SupportedISA"]:
+          tPrint(1, f"# Detected GPU {i}:    {gfxName(arch)}")
+          globalParameters["CurrentISA"] = arch
+
+
 def detectGlobalCurrentISA():
   """
   Returns returncode if detection failure
   """
   global globalParameters
 
-  if globalParameters["CurrentISA"] == (0,0,0) and globalParameters["ROCmAgentEnumeratorPath"]:
-    process = subprocess.run([globalParameters["ROCmAgentEnumeratorPath"]], stdout=subprocess.PIPE)
-    if os.name == "nt":
-      line = ""
-      for line_in in process.stdout.decode().splitlines():
-        if 'gcnArchName' in line_in:
-          line += line_in.split()[1]
-          break # determine if hipinfo will support multiple arch
-      arch = gfxArch(line.strip())
-      if arch is not None:
-        if arch in globalParameters["SupportedISA"]:
-          tPrint(1, "# Detected local GPU with ISA: " + gfxName(arch))
-          globalParameters["CurrentISA"] = arch
-    else:
-      for line in process.stdout.decode().split("\n"):
-        arch = gfxArch(line.strip())
-        if arch is not None:
-          if arch in globalParameters["SupportedISA"]:
-            tPrint(1, "# Detected local GPU with ISA: " + gfxName(arch))
-            globalParameters["CurrentISA"] = arch
-    if (process.returncode):
-      printWarning("%s exited with code %u" % (globalParameters["ROCmAgentEnumeratorPath"], process.returncode))
-    return process.returncode
-  return 0
+  if globalParameters["CurrentISA"] != (0,0,0) or not globalParameters["ROCmAgentEnumeratorPath"]:
+    return 0
```

Comment:
How does this work on windows without rocm agent enumerator?   Do you have windows CI running now?

Reply:
We do not have CI running on Windows. However, since this ticket is aimed at bolstering Windows out-of-the-box configuration, I'm testing all changes on Windows regularly.

Working within the constraints of the current impl, ROCmAgentEnumerator is either set to False (if no enumerate is used) or to the fully qualified path of the device enumerator, `rocm_agent_enumerator` for Linux and `hipInfo.exe` for Windows.